api/app/controllers/spree/api/v2/storefront/wishlists_controller.rb

```diff
@@ -0,0 +1,89 @@
+module Spree
+  module Api
+    module V2
+      module Storefront
+        class WishlistsController < ::Spree::Api::V2::ResourceController
+          include Spree::Api::V2::CollectionOptionsHelpers
+
+          before_action :require_spree_current_user
+
+          def index
+            spree_authorize! :index, Spree::Wishlist
+            wishlists = spree_current_user.wishlists.where(store_id: current_store.id).page(params[:page]).per(params[:per_page])
+
+            render_serialized_payload { serialize_collection(wishlists) }
+          end
+
+          def create
+            spree_authorize! :create, Spree::Wishlist
+            wishlist = Spree::Wishlist.new( wishlist_attributes )
```

Comment:
```suggestion
            wishlist = current_store.wishlists.new( wishlist_attributes )
```

Reply:
@damianlegawiec

**Regarding this Change:** 
I went with creating this through the `spree_current_user ` to ensure the Wishlist is associated with the user, and then used the `ensure_current_store(current_store)` method for store association.

If you think that it is better created through `current_store` and then manually adding the user before saving let me know and I'll switch it up.